flink-core/src/test/java/org/apache/flink/testutils/ClassLoaderUtils.java

```diff
@@ -55,7 +58,7 @@ private static URLClassLoader createClassLoader(File root, ClassLoader parent)
         return new URLClassLoader(new URL[] {root.toURI().toURL()}, parent);
     }
 
-    private static void writeAndCompile(File root, String filename, String source)
+    public static void writeAndCompile(File root, String filename, String source)
```

Comment:
do we need to have it `public`?

Reply:
not related to this change however since this class is touched
I think in this line (above)
```java
return withRoot(root).addClass(filename.replaceAll("\\.java", ""), source).build();
```
we could use just `replace` which doesn't spend resources for regex compilation like 
```java
return withRoot(root).addClass(filename.replace(".java", ""), source).build();
```

Reply:
> do we need to have it public?

Good catch. That is the leftover from my initial approach which I missed to roll back.